server/src/weaverbird/backends/pypika_translator/translators/postgresql.py

```diff
@@ -69,5 +63,18 @@ def duration(
         )
         return StepContext(query, columns + [step.new_column_name])
 
+    @classmethod
+    def _build_unpivot_col(cls, *, step: 'UnpivotStep') -> str:
+        from weaverbird.backends.pypika_translator.translators import RedshiftTranslator
+
+        if issubclass(cls, RedshiftTranslator):
+            return super()._build_unpivot_col(step=step)
+        else:
+            value_col = format_quotes(step.value_column_name, cls.QUOTE_CHAR)
+            unpivot_col = format_quotes(step.unpivot_column_name, cls.QUOTE_CHAR)
+            in_cols = ', '.join(format_quotes(col, cls.QUOTE_CHAR) for col in step.unpivot)
+            in_single_quote_cols = ', '.join(format_quotes(col, "'") for col in step.unpivot)
+            return f' t1 CROSS JOIN unnest(array[{in_single_quote_cols}], array[{in_cols}]) t2 ({unpivot_col}, {value_col})'
+
```

Comment:
Since PostgreSQL and Athena seem to have the same implementation, Why not introduce a new `SUPPORTS_UNPIVOT` attribute in the base translator and use one implem or another depending on the value of the attribute ? And simply set the attribute to `true` in the Redshift translator ? I'm not fond of circular imports :grimacing: